From: Mel Gorman <mgorman@techsingularity.net>
Subject: mm/page_alloc.c: allow __GFP_NOFAIL requests deeper access to reserves
Date: Mon, 9 Jan 2023 15:16:29 +0000

Currently __GFP_NOFAIL allocations without any other flags can access 25%
of the reserves but these requests imply that the system cannot make
forward progress until the allocation succeeds.  Allow __GFP_NOFAIL access
to 75% of the min reserve.

Link: https://lkml.kernel.org/r/20230109151631.24923-6-mgorman@techsingularity.net
Signed-off-by: Mel Gorman <mgorman@techsingularity.net>
Cc: Matthew Wilcox <willy@infradead.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: NeilBrown <neilb@suse.de>
Cc: Thierry Reding <thierry.reding@gmail.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_alloc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/page_alloc.c~mm-page_allocc-allow-__gfp_nofail-requests-deeper-access-to-reserves
+++ a/mm/page_alloc.c
@@ -5323,7 +5323,7 @@ nopage:
 		 * could deplete whole memory reserves which would just make
 		 * the situation worse
 		 */
-		page = __alloc_pages_cpuset_fallback(gfp_mask, order, ALLOC_HARDER, ac);
+		page = __alloc_pages_cpuset_fallback(gfp_mask, order, ALLOC_MIN_RESERVE|ALLOC_HARDER, ac);
 		if (page)
 			goto got_pg;
 
_
